what-changed/model/operation.go

```diff
@@ -4,526 +4,527 @@
+			ExtractStringValueSliceChanges(lOperation.Produces.Value, rOperation.Produces.Value,
+				&changes, v3.ProducesLabel, true)
+		}
+
+		// consumes
+		if len(lOperation.Consumes.Value) > 0 || len(rOperation.Consumes.Value) > 0 {
+			ExtractStringValueSliceChanges(lOperation.Consumes.Value, rOperation.Consumes.Value,
+				&changes, v3.ConsumesLabel, true)
+		}
+
+		// schemes
+		if len(lOperation.Schemes.Value) > 0 || len(rOperation.Schemes.Value) > 0 {
+			ExtractStringValueSliceChanges(lOperation.Schemes.Value, rOperation.Schemes.Value,
+				&changes, v3.SchemesLabel, true)
+		}
+
+		oc.ExtensionChanges = CompareExtensions(lOperation.Extensions, rOperation.Extensions)
+	}
+
+	// OpenAPI
+	if reflect.TypeOf(&v3.Operation{}) == reflect.TypeOf(l) &&
+		reflect.TypeOf(&v3.Operation{}) == reflect.TypeOf(r) {
+
+		lOperation := l.(*v3.Operation)
+		rOperation := r.(*v3.Operation)
+
+		// perform hash check to avoid further processing
+		if low.AreEqual(lOperation, rOperation) {
+			return nil
+		}
+
+		props = append(props, addSharedOperationProperties(lOperation, rOperation, &changes)...)
+		compareSharedOperationObjects(lOperation, rOperation, &changes, oc)
+
+		// parameters
+		lParamsUntyped := lOperation.GetParameters()
+		rParamsUntyped := rOperation.GetParameters()
+		if !lParamsUntyped.IsEmpty() && !rParamsUntyped.IsEmpty() {
+			lParams := lParamsUntyped.Value.([]low.ValueReference[*v3.Parameter])
+			rParams := rParamsUntyped.Value.([]low.ValueReference[*v3.Parameter])
+
+			lv := make(map[string]*v3.Parameter, len(lParams))
+			rv := make(map[string]*v3.Parameter, len(rParams))
+
+			for i := range lParams {
+				s := lParams[i].Value.Name.Value
+				lv[s] = lParams[i].Value
+			}
+			for i := range rParams {
+				s := rParams[i].Value.Name.Value
+				rv[s] = rParams[i].Value
+			}
+
+			var paramChanges []*ParameterChanges
+			for n := range lv {
+				if _, ok := rv[n]; ok {
+					if !low.AreEqual(lv[n], rv[n]) {
+						ch := CompareParameters(lv[n], rv[n])
+						if ch != nil {
+							paramChanges = append(paramChanges, ch)
+						}
+					}
+					continue
+				}
+				CreateChange(&changes, ObjectRemoved, v3.ParametersLabel,
+					lv[n].Name.ValueNode, nil, true, lv[n].Name.Value,
+					nil)
+
+			}
+			for n := range rv {
+				if _, ok := lv[n]; !ok {
+					CreateChange(&changes, ObjectAdded, v3.ParametersLabel,
+						nil, rv[n].Name.ValueNode, true, nil,
+						rv[n].Name.Value)
+				}
+			}
+			oc.ParameterChanges = paramChanges
+		}
+		if !lParamsUntyped.IsEmpty() && rParamsUntyped.IsEmpty() {
+			CreateChange(&changes, PropertyRemoved, v3.ParametersLabel,
+				lParamsUntyped.ValueNode, nil, true, lParamsUntyped.Value,
+				nil)
+		}
+		if lParamsUntyped.IsEmpty() && !rParamsUntyped.IsEmpty() {
+			CreateChange(&changes, PropertyAdded, v3.ParametersLabel,
+				nil, rParamsUntyped.ValueNode, true, nil,
+				rParamsUntyped.Value)
+		}
+
+		// security
+		if !lOperation.Security.IsEmpty() || !rOperation.Security.IsEmpty() {
```

Comment:
Should apply for the other fields as well?

Reply:
Thank you for finding this!

Reply:
There are a few cases where these copy/paste may occur, if you find any more, the same treatment should be applied yes.

Reply:
To be sure we mean the same: `&&` -> `||`?